x/incentives/keeper/distribute.go

```diff
@@ -647,6 +646,39 @@ func (k Keeper) updateGaugePostDistribute(ctx sdk.Context, gauge types.Gauge, ne
 	return nil
 }
 
+// handleGroupPostDistribute handles the post distribution logic for groups and group gauges.
+// If group gauge is perpetual or non-perpetual at the last distribution epoch, it will update the gauge in state by increasing filled epochs 1 and updating
+// the distributed coins field to add the coins distributed.
+// If group gauge is non-perpetual and at the last distribution epoch, it will delete the group and group gauge from state.
+// Before deleting the non-perpetual gauge, any difference between distributed coins and gauge's coins is sent to
+// community pool.
+// CONTRACT: non-perpetual gauge at the last distribution epoch must have distributed all of its coins already
+func (k Keeper) handleGroupPostDistribute(ctx sdk.Context, groupGauge types.Gauge, coinsDistributed sdk.Coins) error {
+	// Prune expired non-perpetual gauges.
+	if groupGauge.IsLastNonPerpetualDistribution() {
+		// Send truncation dust to community pool.
+		truncationDust := groupGauge.Coins.Sub(groupGauge.DistributedCoins.Add(coinsDistributed...))
```

Comment:
Even though it should technically not be possible, I think it would be worthwhile to do a `IsAllGTE` check here to ensure that this dust is never negative so that e.g. weird rounding behavior cannot be escalated to halting the chain

Reply:
Addressed: [702bb34](https://github.com/osmosis-labs/osmosis/pull/6509/commits/702bb3497dfa53ec58d9a18a6c612b85d08224c3)

Reply:
I expanded the check further here: [850b041](https://github.com/osmosis-labs/osmosis/pull/6509/commits/850b0415ff4ed996b1491fc58debe4245c12c766)

The first update was correct as well. However, the last one makes it stricter